pkg/cmd/deploy.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/surajssd/kapp/pkg/encoding"
+	"github.com/surajssd/kapp/pkg/transform/kubernetes"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+)
+
+func Deploy(files []string) error {
+	for _, file := range files {
+
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			return errors.Wrap(err, "file reading failed")
+		}
+
+		app, err := encoding.Decode(data)
+		if err != nil {
+			return errors.Wrap(err, "unable to unmarshal data")
+		}
+
+		ros, err := kubernetes.Transform(app)
+		if err != nil {
+			return errors.Wrap(err, "unable to convert data")
+		}
+
+		for _, o := range ros {
+			data, err := yaml.Marshal(o)
+			if err != nil {
+				return errors.Wrap(err, "failed to marshal object")
+			}
+
+			cmd := exec.Command("kubectl", "create", "-f", "-")
+			stdin, err := cmd.StdinPipe()
+			if err != nil {
+				return errors.Wrap(err, "can't get stdinPipe for kubectl")
+			}
+			go func() {
+				defer stdin.Close()
+				io.WriteString(stdin, string(data[:]))
+			}()
+
+			out, err := cmd.CombinedOutput()
+			if err != nil {
+				return errors.Wrap(err, "failed to execute command")
```

Comment:
In other PR of [e2e tests](https://github.com/surajssd/kapp/pull/77) I created a pipe also for stderr, you can do:

```go
stderr, err := cmd.StdinPipe()
```

see [here](https://github.com/surajssd/kapp/pull/77/files#diff-c4c720d6a8ef6b9eb9238ff4c4deb432R119)

Also before calling `wait()` I [read](https://github.com/surajssd/kapp/pull/77/files#diff-c4c720d6a8ef6b9eb9238ff4c4deb432R138) it.
I am not sure how this will be handled with `CombinedOutput`